src/libtsan.zig

```diff
@@ -43,6 +53,7 @@ pub fn buildTsan(comp: *Compilation, prog_node: std.Progress.Node) BuildError!vo
 
     const optimize_mode = comp.compilerRtOptMode();
     const strip = comp.compilerRtStrip();
+    const link_libcpp = target.isDarwin();
```

Comment:
I suspect this is causing the extra build of libc++ that you are observing. It's not really about the target OS being darwin here, but rather the fact that a dynamic library is being built. A dynamic library is similar to an executable in that its dependency on libc++ must be resolved at link time. Since zig provides libc++ statically, we end up with 1 build of libc++ inside the tsan library, and potentially another inside the application (if the application also links libc++).

Since libc++ is included in libSystem (is that true?) I think it would be better to make dynamic libraries targeting Darwin use the system libc++ rather than the static one. Maybe it would be better to make `-lc++` *always* link system libc++ on Darwin.

Reply:
I completely agree but I would strongly prefer to do this in a subsequent PR. What do you think?